.github/workflows/sync-to-upstream.yml

```
@@ -45,33 +73,35 @@ concurrency:
-  sync-to-apache:
-    name: Sync to Apache Repository
+  sync-to-upstream:
+    name: Sync to Upstream Repository
     runs-on: ubuntu-latest
     
-    # Only run when PR is actually merged (not just closed)
-    if: github.event.pull_request.merged == true
+    # Only run when:
+    # 1. PR is actually merged (not just closed)
+    # 2. PR targets the base branch configured in env.BASE_BRANCH
+    if: github.event.pull_request.merged == true && github.event.pull_request.base.ref == env.BASE_BRANCH
```

actionlint (1.7.7)

context "env" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)